tests/unit/framework/test_pipeline_manager.py

```diff
@@ -314,3 +392,229 @@ def test_throttle_sleep_time_increases_with_qsize(self):
+        target = "store"
+        output_config = {"random_name": {"type": "dummy_output"}}
+        queue = ThrottlingQueue(multiprocessing.get_context(), 100)
+        queue.empty = mock.MagicMock(return_value=True)
+        listener = ComponentQueueListener(queue, target, output_config)
+        with mock.patch("logprep.connector.dummy.output.DummyOutput.setup") as mock_setup:
+            listener.queue.put("test")
+            listener.queue.put(listener.sentinel)
+            listener._listen()
+        mock_setup.assert_called()
+
+    def testget_component_instance_raises_if_setup_not_successful(self):
+        target = "store"
+        output_config = {"random_name": {"type": "dummy_output"}}
+        queue = ThrottlingQueue(multiprocessing.get_context(), 100)
+        listener = ComponentQueueListener(queue, target, output_config)
+        with mock.patch("logprep.connector.dummy.output.DummyOutput.setup") as mock_setup:
+            mock_setup.side_effect = SystemExit(4)
+            with pytest.raises(SystemExit, match="4"):
+                listener._listen()
+
+    def test_listen_calls_component_shutdown_by_injecting_sentinel(self):
+        target = "store"
+        output_config = {"random_name": {"type": "dummy_output"}}
+        queue = ThrottlingQueue(multiprocessing.get_context(), 100)
+        queue.empty = mock.MagicMock(return_value=True)
+        listener = ComponentQueueListener(queue, target, output_config)
+        with mock.patch("logprep.connector.dummy.output.DummyOutput.shut_down") as mock_shutdown:
+            listener.queue.put("test")
+            listener.queue.put(listener.sentinel)
+            listener._listen()
+        mock_shutdown.assert_called()
+
+    @mock.patch(
+        "logprep.framework.pipeline_manager.ComponentQueueListener.get_component_instance",
+        new=mock.MagicMock(),
+    )
+    def test_listen_drains_queue_on_shutdown(self):
+        target = "store"
+        output_config = {"random_name": {"type": "dummy_output"}}
+        queue = multiprocessing.Queue()
+        listener = ComponentQueueListener(queue, target, output_config)
+        listener.queue.put(listener.sentinel)
+        listener.queue.put("test")
+        listener._listen()
+        assert listener.queue.qsize() == 0
+
+    @mock.patch(
+        "logprep.framework.pipeline_manager.ComponentQueueListener.get_component_instance",
+        new=mock.MagicMock(),
+    )
+    def test_listen_ensures_error_queue_is_closed_after_drained(self):
+        target = "store"
+        output_config = {"random_name": {"type": "dummy_output"}}
+        queue = ThrottlingQueue(multiprocessing.get_context(), 100)
+        listener = ComponentQueueListener(queue, target, output_config)
+        random_number = random.randint(10, 50)
+        listener.queue.put(listener.sentinel)
+        for _ in range(random_number):
+            listener.queue.put("test")
+        while not listener.queue.qsize() == random_number + 1:
+            pass
```

Comment:
Could probably be removed, test putting something into the queue only after the `_listen`.